ivy/functional/backends/numpy/linear_algebra.py

```diff
@@ -31,6 +31,40 @@ def cholesky(
     return ret
 
 
+@with_unsupported_dtypes({"1.23.0 and below": ("float16",)}, backend_version)
+def cov(
+    x1: np.ndarray,
+    x2: np.ndarray = None,
+    /,
+    *,
+    rowVar: bool = True,
+    bias: bool = False,
+    ddof: int = None,
+    fweights: np.ndarray = None,
+    aweights: np.ndarray = None,
+    dtype: Optional[np.dtype] = None,
+    out: Optional[np.ndarray] = None,
+) -> np.ndarray:
+    x1, x2 = ivy.promote_types_of_inputs(x1, x2)
+
+    if fweights is not None:
```

Comment:
once the tests are configured correctly, implicit casting here shouldn't be necessary! unless it's related to the numpy implicit casting rule, you may have to tweak the tests a little to make sure array or dtype combinations follow the native conventions!